contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -369,9 +314,20 @@ contract RibbonTreasuryVault is
      */
     function setCap(uint256 newCap) external onlyOwner {
         require(newCap > 0, "!newCap");
+
+        uint256 previousCap = vaultParams.cap;
+        vaultParams.cap = _getCap104(newCap);
+
+        emit CapSet(previousCap, newCap);
+    }
+
+    /**
+     * @notice Internal function to convert cap into uint104
+     * @param newCap is the cap to convert into uint104
+     */
+    function _getCap104(uint256 newCap) internal pure returns (uint104) {
         ShareMath.assertUint104(newCap);
-        emit CapSet(vaultParams.cap, newCap);
-        vaultParams.cap = uint104(newCap);
+        return uint104(newCap);
```

Comment:
Could just refactor this functionality into `ShareMath` as the function `ShareMath.toUint104`. Does not have to be specific to `cap`. See how we do it for https://github.com/ribbon-finance/governance/blob/main/contracts/rbn-staking/IncentivizedVotingLockup.sol#L835

Reply:
Reverting this to the initial version.

```
function setCap(uint256 newCap) external onlyOwner {
      require(newCap > 0, "!newCap");
      ShareMath.assertUint104(newCap);
      emit CapSet(vaultParams.cap, newCap);
      vaultParams.cap = uint104(newCap);
  }
```


This change was reflected here because it was not merged with the fee-distribution branch yet.